src/components/LogIn/LogIn.jsx

```diff
@@ -0,0 +1,225 @@
+import axios from "axios";
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useAuthContext } from "../../global-context/auth-context";
+import { useAlertTextContext } from "../../global-context/alert-text-context";
+import AlertText from "../AlertText/AlertText";
+
+const LogIn = () => {
+  const { alertText, setAlertText } = useAlertTextContext();
+
+  const [loginFormData, setLoginFormData] = useState({
+    userEmail: "",
+    userPassword: "",
+    persistUser: true,
+    showPassword: false,
+  });
+  const { userEmail, userPassword, persistUser, showPassword } = loginFormData;
+  const navigateTo = useNavigate();
+  const { setIsUserAuthenticated } = useAuthContext();
+
+  const logInService = async (email, password) => {
+    try {
+      const resp = await axios.post("/api/auth/login", {
+        email,
+        password,
+      });
+
+      return (
+        (resp.status === 200 || resp.status === 201) && resp.data.encodedToken
+      );
+    } catch (error) {
+      console.log(error);
+      setAlertText((prevObj) => ({
+        ...prevObj,
+        isActive: true,
+        message:
+          error.response.status === 404
+            ? "Incorrect username or password"
+            : "Something went wrong",
+        alertType: "danger",
+      }));
+    }
+  };
+
+  const userLoginHandler = async (userEmail, userPassword, persistUser) => {
+    const encToken = await logInService(userEmail, userPassword);
+    let from = location.state?.from?.pathname || "/";
+
+    try {
+      if (encToken) {
+        persistUser
+          ? localStorage.setItem("notesToken", encToken)
+          : sessionStorage.setItem("notesToken", encToken);
+        console.log(from);
+        setIsUserAuthenticated(true);
+
+        navigateTo(from, { replace: true });
+      }
+    } catch (error) {
+      console.log(error); //ensure proper error handling
+    }
+  };
+
+  return (
+    <div className="modal-container flex-center" id="modal-container-js">
+      <div className="modal-content padding-mdm margin-mdm">
+        <button
+          className="btn btn-outlined btn-sml pos-abs-top-right box-shadow-none red-text-color"
+          id="btn-close-modal"
+          onClick={() => navigateTo(-1)}
```

Comment:
Want to understand its working as to what it does!!But got to know that it is going one step back

Reply:
Yeah as tanay mentioned it is just a hack not an authentic solution